dht.go

```diff
@@ -73,7 +78,11 @@ type IpfsDHT struct {
 
 	stripedPutLocks [256]sync.Mutex
 
-	protocols []protocol.ID // DHT protocols
+	// Primary DHT protocols - we query and respond to these protocols
+	protocols []protocol.ID
+
+	// DHT protocols we can respond to (may contain protocols in addition to the primary protocols)
+	serverProtocols []protocol.ID
```

Comment:
We probably also need `kad1Protocol` and `kad2Protocol` for figuring out how we should respond to peers.

Reply:
SGTM, we can tackle that once we actually have different protocol handlers to use (hopefully won't be too long 🙏)